pepper-apis/src/main/java/org/broadinstitute/ddp/event/pubsubtask/api/PubSubTaskConnectionService.java

```
@@ -0,0 +1,139 @@
+
+    private PubSubTaskReceiver pubSubTaskReceiver;
+    private PubSubTaskResultSender pubSubTaskResultSender;
+
+    private Subscriber pubSubTaskSubscriber;
+    private Publisher pubSubTaskResultPublisher;
+
+
+    public PubSubTaskConnectionService(
+            PubSubConnectionManager pubSubConnectionManager,
+            String projectId,
+            String subscription,
+            String pubSubTaskResultPubSubTopic,
+            int subscriberAwaitRunningTimeout,
+            PubSubTaskProcessorFactory pubSubTaskProcessorFactory) {
+        this.pubSubConnectionManager = pubSubConnectionManager;
+        projectSubscriptionName = ProjectSubscriptionName.of(projectId, subscription);
+        pubSubTaskResultProjectTopicName = ProjectTopicName.of(projectId, pubSubTaskResultPubSubTopic);
+        this.subscriberAwaitRunningTimeout = subscriberAwaitRunningTimeout;
+        this.pubSubTaskProcessorFactory = pubSubTaskProcessorFactory;
+    }
+
+    public void create() {
+        try {
+            createPubSubTaskResultPublisher();
+            pubSubTaskResultSender = new PubSubTaskResultSender(pubSubTaskResultPublisher);
+            pubSubTaskReceiver = new PubSubTaskReceiver(projectSubscriptionName, pubSubTaskProcessorFactory, pubSubTaskResultSender);
+            createPubSubTaskSubscriber();
+        } catch (Exception e) {
+            LOG.error(errorMsg("Failed to create PubSubTask connection"), e);
+        }
+    }
+
+    public void destroy() {
+        if (pubSubTaskSubscriber != null && pubSubTaskSubscriber.isRunning()) {
+            pubSubTaskSubscriber.stopAsync();
+        }
+        if (pubSubTaskResultPublisher != null) {
+            try {
+                pubSubTaskResultPublisher.shutdown();
+                pubSubTaskResultPublisher.awaitTermination(PUBLISHER_AWAIT_TERMINATION_TIMEOUT_MIN, TimeUnit.MINUTES);
+            } catch (Exception e) {
+                LOG.error(errorMsg("Failed to shutdown PubSubTask connection"), e);
```

Seems like this error and the one on L76 are fatal--perhaps throw a runtime exception instead of logging?

---

Ok. But seems it will be fatal for whole Housekeeping. Is it ok?

---

How about wrapping this in a try/catch when it's called in `Housekeeping.java`?

---

Fixed